### PR TITLE
Skills add, delete, name edit, error handling

### PR DESCRIPTION
**Pull request by urrri**
#16 #17

### CODE REVIEW

File: src/components/Skills.js [hunk 1]
```diff
@@ -1,7 +1,17 @@
 import React, { Component, PropTypes } from 'react';
 import autoBind from 'react-autobind';
-import classNames from 'classnames';
 import _ from 'lodash';
+import EditToggle from './EditToggle';
+
+let uidGen = 1;
```
Comment (assafg) [inline src/components/Skills.js:6]: What is this counter for? Skill name is unique

**Comment by urrri** [comment]
Skill name is editable thing so it cannot be unique. Any user editable
field cannot be used as key.

**Comment by urrri** [comment]
As for uniquness of the Skill name I verify it on edit and disable
save on any error

**Comment by assafg** [comment]
Skill name MUST stay unique sine it is a key... It shold be validated prior
to changing.

**Comment by urrri** [comment]
It is validating, but... Let's imagine we have entry Java and now entering
JavaScript. When we reached Java part of JavaScript we can mark it as
duplicated and prevent saving or we can disallow user to enter second 'a',
because it changes key to non-unique. I prefer notifying user and prevent
saving, because second behavior is very unfriendly.

In addition we have some other error states: empty skill name, the same skill name in different cases. I catch them all

**Pull request merged**